Vector.hpp

```diff
@@ -0,0 +1,246 @@
+#ifndef VECTOR_HPP
+
+#define VECTOR_HPP
+
+#include <iostream>
+
+
+template<typename T>
+class Vector
+{
+public:
+	/**
+	 * @brief Default constructor.
+	 *
+	 * Constructs an empty Vector with an initial capacity of 2.
+	 */
+	Vector()
+	{
+		Realloc(2);
+	}
+	/**
+	 * @brief Destructor.
+	 *
+	 * Frees the dynamically allocated memory.
+	 */
+	~Vector()
+	{
+
+		delete(_dynamicVector);
+	}
+
+	/*@brief Adds an element to the end of the vector.
+	*
+     **/
+	void pushBack(const T& value)
+		
+	{
+		if (_size >= _capacity)
+		{
+			Realloc(_capacity + _capacity / 2);
+
+		}
+		_dynamicVector[_size] = value;
+		_size++;
+	}
+
+	/*display the Vector*/
+	void display() const
+	{
+		for (size_t i = 0; i < _size; ++i) {
+			std::cout << _dynamicVector[i] << " ";
+		}
+		std::cout << std::endl;
+	}
+
+	/*@return returns the element in index i
+	@throw  std::out_of_range if the index is out of range*/
+	T& at(int i) const
+	{
+		if (i < 0 || i >= _size)
+		{
+			throw std::out_of_range("Index is out range");
+		}
+		return _dynamicVector[i];
+	}
+
+	/*@return returns the element in index i*/
+	T& operator[](int i)
```

Comment:
i would suggest adding checks for the validity of the input.
also are you sure an int should be used here?